kpi/tests/test_assets.py

```diff
@@ -432,6 +432,18 @@ def test_templates_move_form_title_to_name(self):
         self.assertTrue('form_title' not in settings)
         self.assertEqual(a1.name, 'abcxyz')
 
+    def test_validate_asset_settings(self):
+        unsanitized_html_str = "<h1>HTML Injection testing</h1>"
+        _content = self._content()
+        _content['survey'][0]['default'] = unsanitized_html_str
+
+        def _create_asset():
+            Asset.objects.create(
+                content=_content,
+                owner=self.user,
+                asset_type='survey')
+        self.assertRaises(ValidationError, _create_asset)
```

Comment:
1. Should we assert that the validation error message is as expected?
2. Is the case where an error is not raised already tested?

Reply:
Ive added these test scenerios